grpc_opa/obligations_test.go

```diff
@@ -0,0 +1,371 @@
+						&ObligationsNode{
+							Kind: ObligationsCondition,
+							Condition: "c",
+						},
+					},
+				},
+				&ObligationsNode{
+					Kind: ObligationsOr,
+					Children: []*ObligationsNode{
+						&ObligationsNode{
+							Kind: ObligationsCondition,
+							Condition: "a",
+						},
+						&ObligationsNode{
+							Kind: ObligationsCondition,
+							Condition: "b",
+						},
+					},
+				},
+			},
+		},
+	},
+	{
+		expectedErr:  nil,
+		regoRespJSON: `{
+			"allow": true,
+			"obligations": {}
+		}`,
+		expectedVal:  &ObligationsNode{},
+	},
+	{
+		expectedErr:  nil,
+		regoRespJSON: `{
+			"allow": true,
+			"obligations": {
+				"policy1_guid": {},
+				"policy2_guid": {}
+			}
+		}`,
+		expectedVal:  &ObligationsNode{},
+	},
+	{
+		expectedErr:  nil,
+		regoRespJSON: `{
+			"allow": true,
+			"obligations": {
+				"policy1_guid": {},
+				"policy2_guid": {
+					"stmt1": []
+				}
+			}
+		}`,
+		expectedVal:  &ObligationsNode{},
+	},
+	{
+		expectedErr:  nil,
+		regoRespJSON: `{
+			"allow": true,
+			"obligations": {
+				"policy1_guid": {
+					"stmt0": []
+				},
+				"policy2_guid": {
+					"stmt1": []
+				}
+			}
+		}`,
+		expectedVal:  &ObligationsNode{},
+	},
+	{
+		expectedErr:  nil,
+		regoRespJSON: `{
+			"allow": true,
+			"obligations": {
+				"policy1_guid": {
+					"stmt0": [ "i", "j" ]
+				},
+				"policy2_guid": {}
+			}
+		}`,
+		expectedVal:  &ObligationsNode{
+			Kind: ObligationsOr,
+			Children: []*ObligationsNode{
+				&ObligationsNode{
+					Kind: ObligationsOr,
+					Tag: "policy1_guid",
+					Children: []*ObligationsNode{
+						&ObligationsNode{
+							Kind: ObligationsOr,
+							Tag: "stmt0",
+							Children: []*ObligationsNode{
+								&ObligationsNode{
+									Kind: ObligationsCondition,
+									Condition: "i",
+								},
+								&ObligationsNode{
+									Kind: ObligationsCondition,
+									Condition: "j",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	},
+	{
+		expectedErr:  nil,
+		regoRespJSON: `{
+			"allow": true,
+			"obligations": {
+				"policy1_guid": {
+					"stmt0": [ "i", "j", "k" ]
+				},
+				"policy2_guid": {
+					"stmt1": []
+				}
+			}
+		}`,
+		expectedVal:  &ObligationsNode{
+			Kind: ObligationsOr,
+			Children: []*ObligationsNode{
+				&ObligationsNode{
+					Kind: ObligationsOr,
+					Tag: "policy1_guid",
+					Children: []*ObligationsNode{
+						&ObligationsNode{
+							Kind: ObligationsOr,
+							Tag: "stmt0",
+							Children: []*ObligationsNode{
+								&ObligationsNode{
+									Kind: ObligationsCondition,
+									Condition: "i",
+								},
+								&ObligationsNode{
+									Kind: ObligationsCondition,
+									Condition: "j",
+								},
+								&ObligationsNode{
+									Kind: ObligationsCondition,
+									Condition: "k",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	},
+	{
+		expectedErr:  nil,
+		regoRespJSON: `{
+			"allow": true,
+			"obligations": {
+				"policy1_guid": {
```

Comment:
Are we returning policies in obligations? I thought it was just metadata they send through authz. @wplu can you review this PR?

Reply:
This is the metadata that is being returned, to make it easier to debug which policy(s) contributed the obligations. @daniel-garcia was also in the discussion about this.